homeassistant/components/backup/manager.py

```diff
@@ -180,23 +211,23 @@ async def get_backup(self, slug: str) -> Backup | None:
 
         return backup
 
-    async def remove_backup(self, slug: str) -> None:
+    async def async_remove_backup(self, *, slug: str, **kwargs: Any) -> None:
         """Remove a backup."""
-        if (backup := await self.get_backup(slug)) is None:
+        if (backup := await self.async_get_backup(slug=slug)) is None:
             return
 
         await self.hass.async_add_executor_job(backup.path.unlink, True)
         LOGGER.debug("Removed backup located at %s", backup.path)
         self.backups.pop(slug)
 
-    async def generate_backup(self) -> Backup:
+    async def async_create_backup(self, **kwargs: Any) -> Backup:
         """Generate a backup."""
         if self.backing_up:
             raise HomeAssistantError("Backup already in progress")
 
         try:
             self.backing_up = True
-            await self.pre_backup_actions()
+            await self.async_pre_backup_actions()
```

Comment:
It feels like the abstract interface allows a flaw in our model since backup managers can decide to not call the pre and post back actions when creating a backup. What's our view on this?

Reply:
Even with this base class, on a supervised system the supervisor will create the backup of the core instance.
These (async_pre_backup_actions and async_post_backup_actions) should only be executed during backup creation for core/container, for supervised/OS they should be executed with the dedicated WS commands.

Reply:
Ok